src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/scope_map.py

```diff
@@ -0,0 +1,159 @@
+    if not validated:
+        raise CLIError("Rule {} has invalid syntax.".format(actions))
+    actions.sort()
+
+    if description is not None:
+        description = ' '.join(description)
+
+    resource_group_name = get_resource_group_name_by_registry_name(cmd, registry_name, resource_group_name)
+
+    from msrest.exceptions import ValidationError
+    try:
+        return client.create(
+            resource_group_name,
+            registry_name,
+            scope_map_name,
+            actions,
+            description
+        )
+    except ValidationError as e:
+        raise CLIError(e)
+
+
+def acr_scope_map_delete(cmd,
+                         client,
+                         registry_name,
+                         scope_map_name,
+                         resource_group_name=None):
+
+    from knack.prompting import prompt_y_n
+    confirmation = prompt_y_n("Deleting the scope map '{}' will remove its permissions with associated tokens. Are you sure you want to proceed?")
+
+    if confirmation in ['N', 'n']:
+        return
+
+    resource_group_name = get_resource_group_name_by_registry_name(cmd, registry_name, resource_group_name)
+    return client.delete(resource_group_name, registry_name, scope_map_name)
+
+
+def acr_scope_map_update(cmd,
+                         client,
+                         registry_name,
+                         scope_map_name,
+                         add_repository=None,
+                         remove_repository=None,
+                         reset_map=None,
+                         resource_group_name=None,
+                         description=None):
+
+    if not (add_repository or remove_repository or reset_map or description):
+        raise CLIError("At least one of the following parameters must be provided: " +
+                       "--add, --remove, --reset, --description.")
+
+    current_scope_map = acr_scope_map_show(cmd, client, registry_name, scope_map_name, resource_group_name)
+
+    if reset_map:
+        current_actions = []
+    else:
+        current_actions = current_scope_map.actions
+
+    if description is None:
+        description = current_scope_map.description
+    else:
+        description = ' '.join(description)
+
+    if remove_repository is not None:
+        validated, removed_actions = _validate_and_generate_actions_for_repositories(remove_repository)
+        if not validated:
+            raise CLIError("Rule {} has invalid syntax.".format(removed_actions))
+        current_actions = list(set(current_actions) - set(removed_actions))
+
+    if add_repository is not None:
+        validated, added_actions = _validate_and_generate_actions_for_repositories(add_repository)
+        if not validated:
+            raise CLIError("Rule {} has invalid syntax.".format(added_actions))
+        current_actions = list(set(current_actions) | set(added_actions))
+
+    current_actions.sort()
```

Comment:
curious as to what does it sort on?

Reply:
It sorts the actions in lexicographical order.
However this can still cause issues, as actions can blend with repo name.
The best here is to do a custom sort which parses repos and actions and sort by repos.
I will try to do this